main.tf

```diff
@@ -763,6 +765,66 @@ data "aws_iam_policy_document" "access_log_delivery" {
   }
 }
 
+# Grant read and list access to CloudFront Origin Access Identities (deprecated by AWS)
+data "aws_iam_policy_document" "cloudfront_oai_read_policy" {
+  count = local.create_bucket && var.attach_cloudfront_oai_read_policy ? 1 : 0
+
+  statement {
+    sid = "AWSCloudFrontOAIRead"
+    actions   = ["s3:GetObject"]
+    resources = ["${aws_s3_bucket.this[0].arn}/*"]
+    principals {
+      type        = "AWS"
+      identifiers = var.cloudfront_oai_iam_arns
+    }
+  }
+
+  statement {
+    sid = "AWSCloudFrontOAIList"
+    actions   = ["s3:ListBucket"]
```

Comment:
`s3:ListBucket` doesn't seem to be needed according to the docs - https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html#private-content-restricting-access-to-s3-oai